### PR TITLE
ci: replace home-grow `cargo semver-checks` action

### PR DESCRIPTION
**Pull request by thomaseizinger**
## Description

This PR replaces our home-grown `cargo semver-checks` action with the official one: https://github.com/obi1kenobi/cargo-semver-checks-action/issues/47.

This action has support for cargo workspaces and will automatically check all crates that have `publish` set to `true`. This removes a big time component for our CI checks and introduces one that runs in parallel with all others. I hope to reduce the effective CI time through that, plus it means we need to maintain less code.

The official action would download the latest release of cargo-semver-checks. This might result in a non-reproducible CI. To avoid this, we retain the one-line download statement from our home-grown action.

## Notes & open questions

~This is currently blocked by https://github.com/obi1kenobi/cargo-semver-checks-action/issues/47~

## Change checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -245,6 +231,12 @@ jobs:
             cargo check --manifest-path "$toml";
           done
 
+  semver:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: obi1kenobi/cargo-semver-checks-action@v2
```
Comment (thomaseizinger) [inline .github/workflows/ci.yml:268]:
```suggestion
      - uses: thomaseizinger/cargo-semver-checks-action@060d5a2882a4b9ec99b054efb1b13115acc127ce
        with:
          version: "latest"
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -245,6 +231,14 @@ jobs:
             cargo check --manifest-path "$toml";
           done
 
+  semver:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: thomaseizinger/cargo-semver-checks-action@060d5a2882a4b9ec99b054efb1b13115acc127ce
+        with:
+          version: "latest"
```
Comment (thomaseizinger) [inline .github/workflows/ci.yml]:
```suggestion
          version: "v0.20.1"
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -245,6 +231,14 @@ jobs:
...
           done
 
+  semver:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: thomaseizinger/cargo-semver-checks-action@060d5a2882a4b9ec99b054efb1b13115acc127ce
+        with:
+          version: "v0.20.1"
+
```
Comment (thomaseizinger) [inline .github/workflows/ci.yml]:
```suggestion
      - uses: thomaseizinger/cargo-semver-checks-action@b266e63c6c2823106f61fd9ee4805ac6ce0a2783
        with:
          version: "foobar"

```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -271,6 +257,14 @@ jobs:
             cargo check --manifest-path "$toml";
           done
 
+  semver:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: thomaseizinger/cargo-semver-checks-action@b266e63c6c2823106f61fd9ee4805ac6ce0a2783
+        with:
+          version: "foobar"
```
Comment (thomaseizinger) [inline .github/workflows/ci.yml]:
```suggestion
          version: "baz"
```

**Comment by thomaseizinger** [comment]
Marking this as trivial because it is just a refactoring and we have a lot of spurious CI errors from `cargo semver-checks` when it is downloading from crates.io. With the increase in the number of jobs it is no longer viable to run `cargo semver-checks` for every crate. Plus now that the official action exists, we can just run it for the entire workspace.

**Comment by thomaseizinger** [comment]
@mxinden Please still review this.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -273,6 +259,14 @@ jobs:
             cargo check --manifest-path "$toml";
           done
 
+  semver:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - run: wget -q -O- https://github.com/obi1kenobi/cargo-semver-checks/releases/download/v0.22.1/cargo-semver-checks-x86_64-unknown-linux-gnu.tar.gz | tar -xz -C ~/.cargo/bin
```
Comment (mxinden) [inline .github/workflows/ci.yml:266]: Why is this needed? Doesn't `obi1kenobi/cargo-semver-checks-action@v2` bring `cargo-semver-checks` along with it? Does this speed up CI?

Comment (thomaseizinger) [reply .github/workflows/ci.yml:266]: See https://github.com/obi1kenobi/cargo-semver-checks-action/issues/47.

**Pull request merged**